include/rtc/chaininterop.hpp

```diff
@@ -0,0 +1,82 @@
+#ifndef RTC_CHAIN_INTEROP_H
+#define RTC_CHAIN_INTEROP_H
+
+#if RTC_ENABLE_MEDIA
+#include "common.hpp"
+#include <map>
+#include <deque>
+#include <vector>
+#include <chrono>
+// ChainInterop will be used for storing info about TWCC packets
+// RtpTwccHandler produces new entries inside ChainInterop as it is processing RTP packets.
+// GCC algorithm consumes it.
+namespace rtc {
+	struct RTC_CPP_EXPORT BitrateStats {
+		double txBitsPerSecond, rxBitsPerSecond, packetLoss;
+		BitrateStats() : txBitsPerSecond(0), rxBitsPerSecond(0), packetLoss(0) {}
+		BitrateStats(double tx, double rx, double loss) : txBitsPerSecond(tx), rxBitsPerSecond(rx), packetLoss(loss) {};
+	};
+	
+	struct PacketInfo {
+		bool isReceived;
+		bool isSent;
+		uint16_t numBytes;
+		std::chrono::microseconds arrivalDuration;
+		std::chrono::steady_clock::time_point departureTime;
+		PacketInfo(uint16_t numBytes);
+	};
+	// WholeFrameInfo is an internal representation for a video frame
+	class WholeFrameInfo {
+		std::chrono::steady_clock::time_point time;
+		uint16_t seqNumStart, seqNumEnd;
+	
+	public:
+		WholeFrameInfo(std::chrono::steady_clock::time_point time, uint16_t seqNumStart, uint16_t seqNumEnd);
+		std::chrono::steady_clock::time_point getTime() const;
+		uint16_t getSeqNumStart() const;
+		uint16_t getSeqNumEnd() const;
+	};
+	
+	struct RTC_CPP_EXPORT ArrivalGroup {
+		std::vector<PacketInfo> packets;
+		std::chrono::microseconds arrivalTime;
+		std::chrono::steady_clock::time_point departureTime;
+	
+		ArrivalGroup() = default;
+		ArrivalGroup(const ArrivalGroup &other);
+		ArrivalGroup(ArrivalGroup&& other) noexcept;
+		ArrivalGroup& operator=(const ArrivalGroup& other);
+		ArrivalGroup& operator=(ArrivalGroup&& other) noexcept;
+		void add(PacketInfo packet);	
+		void reset();
+	};
+	
+	class RTC_CPP_EXPORT ChainInterop {
```

Comment:
I'm not a fan of the `ChainInterop` name as it doesn't really say anything about what state the class keeps or what it does.

Also, is this an API you want to expose to the user? It seems to me the user should only access a high-level rate controller interface and the internal logic and helper objects should be somewhere in `impl`.